mitodl/open-discussions
# refactor the tooltips!

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #1299
closes #2046
closes #2209

#### What's this PR do?

This refactors our LoginPopup and SharePopup to be LoginTooltip and ShareTooltip, and tears out our homemade stuff for showing these popups, to use a library instead (this one: https://react-component.github.io/tooltip/)

Basically, now you can just grab the `<LoginTooltip>` and wrap it around any element you want to trigger a popup when clicked, like this:

```js
<LoginTooltip>
  <div className="authenticated-users-only">
    FUN!
  </div>
</LoginTooltip>
```

and that's it! No messing with the store, no nothing.

#### How should this be manually tested?

Make sure that:

- there are no regressions in the things which currently have a login tooltip on them. Like, clicking on anything that requires the user to be logged in on the PostPage, for instance, should trigger a tooltip if you arent logged in and should do the normal thing if you are logged in
- the "favorite" button on the learning resource card is now something that also triggers a login tooltip
- the "share" popup on the post page (for the post and for the comments) works as it did before

I think that's it. Probably a good idea to read through all the code and so on too :smile: 

#### Screenshots (if appropriate)

![newloginsearch](https://user-images.githubusercontent.com/6207644/66341827-01c8cd00-e916-11e9-9c1b-446f2fa3c571.png)
![newlogin](https://user-images.githubusercontent.com/6207644/66341828-01c8cd00-e916-11e9-9387-7d0d0ebabc5d.png)
![newshare](https://user-images.githubusercontent.com/6207644/66341829-01c8cd00-e916-11e9-8bed-54542ef4694b.png)
